arches/app/media/js/viewmodels/workflow-step.js

```diff
@@ -37,6 +37,15 @@ define([
         this.postSaveCallback = ko.observable();
 
         this.externalStepData = {};
+        this.locked = ko.observable();
+
+        this.setLock = function(stepName, locked) {
```

Comment:
A few thing here:

- Can this function declaration be moved to below initialize where the other func declarations are?
- As written, this function offers too much flexibility down the stack. Can it be refactored to not take any step name, and instead rely on a `config` value?
- Can the func name be changed to something more explicit? `setLock` sounds localized on this level and this deals with external steps